src/components/ReanimatedDrawerLayout.tsx

```diff
@@ -0,0 +1,750 @@
+                  -1 * sideCorrectedDragX.value,
+                  [drawerWidth - 1, drawerWidth, drawerWidth + 1],
+                  [0, 0, 1]
+                )
+              : sideCorrectedDragX.value;
+        })
+        .onEnd(handleRelease);
+    }, [
+      dragX,
+      drawerLockMode,
+      drawerTranslation,
+      drawerType,
+      drawerWidth,
+      emitStateChanged,
+      gestureOrientation,
+      handleRelease,
+      edgeHitSlop,
+      fillHitSlop,
+      minSwipeDistance,
+      props.hideStatusBar,
+      props.keyboardDismissMode,
+      sideCorrectedDragX.value,
+      overlayDismissGesture,
+      drawerOpened,
+      isFromLeft,
+      containerWidth,
+      sideCorrection,
+      drawerState,
+      activeCursor,
+      enableTrackpadTwoFingerGesture,
+      mouseButton,
+      statusBarAnimation,
+    ]);
+
+    // When using RTL, row and row-reverse flex directions are flipped.
+    const reverseContentDirection = I18nManager.isRTL
+      ? isFromLeft
+      : !isFromLeft;
+
+    const dynamicDrawerStyles = {
+      backgroundColor: drawerBackgroundColor,
+      width: drawerWidth,
+    };
+
+    const containerStyles = useAnimatedStyle(() => {
+      if (drawerType === DrawerType.FRONT) {
+        return {};
+      }
+
+      return {
+        transform: [
+          {
+            translateX: interpolate(
+              openValue.value,
+              [0, 1],
+              [0, drawerWidth * sideCorrection],
+              Extrapolation.CLAMP
+            ),
+          },
+        ],
+      };
+    });
+
+    const drawerAnimatedStyle = useAnimatedStyle(() => {
+      const closedDrawerOffset = drawerWidth * -sideCorrection;
+      const isBack = drawerType === DrawerType.BACK;
+      const isIdle = drawerState === DrawerState.IDLE;
+
+      if (isBack) {
+        return {
+          transform: [{ translateX: 0 }],
+          flexDirection: reverseContentDirection ? 'row-reverse' : 'row',
+        };
+      }
+
+      let translateX = 0;
+
+      if (isIdle) {
+        translateX = drawerOpened ? 0 : closedDrawerOffset;
+      } else {
+        translateX = interpolate(
+          openValue.value,
+          [0, 1],
+          [closedDrawerOffset, 0],
+          Extrapolation.CLAMP
+        );
+      }
+
+      return {
+        transform: [{ translateX }],
+        flexDirection: reverseContentDirection ? 'row-reverse' : 'row',
+      };
+    });
+
+    const containerAnimatedProps = useAnimatedProps(() => ({
+      importantForAccessibility:
+        Platform.OS === 'android'
+          ? isDrawerOpen.value
+            ? ('no-hide-descendants' as const)
+            : ('yes' as const)
+          : undefined,
+    }));
+
+    const children =
+      typeof props.children === 'function'
+        ? props.children(openValue) // renderer function
+        : props.children;
+
+    const componentRef = useRef(null);
+
+    useImperativeHandle(
+      ref,
+      () => ({
+        ...componentRef,
```

Comment:
The goal here it to add a `current` field to the ref?

Reply:
This was a mistake, it didn't reveal any additional props to the user.
Removed in [0972256](https://github.com/software-mansion/react-native-gesture-handler/pull/3146/commits/09722561121239b19f28186f4c92fd200f3b7975)